### PR TITLE
Support rendering LinkNodes

### PR DESCRIPTION
**Pull request by greglandrum**
These are just another type of bracket, which we already know how to do.
Examples:
image

image

### CODE REVIEW

File: Code/GraphMol/MolDraw2D/MolDraw2D.cpp [hunk 1]
```diff
@@ -2328,6 +2330,70 @@ void MolDraw2D::extractRadicals(const ROMol &mol) {
...
+      std::vector<std::pair<Point2D, Point2D>> bondSegments;  // not needed here
+      MolDrawShape shp;
+      shp.points =
+          MolDraw2D_detail::getBracketPoints(p1, p2, startLoc, bondSegments);
+      shp.shapeType = MolDrawShapeType::Polyline;
+      post_shapes_[activeMolIdx_].emplace_back(std::move(shp));
+
+      if (p1.x > labelPt.x) {
+        labelPt = p1;
+        labelPerp = crossingPt - startLoc;
```
Comment (DavidACosgrove) [inline Code/GraphMol/MolDraw2D/MolDraw2D.cpp:2372]: I haven't followed the logic through completely, but it seems to me that either this line could be outside the condition, and line 2376 likewise, or there's a potential for a division by zero error at line 2386.  Or, more likely, I've missed something obvious.

File: Code/GraphMol/MolEnumerator/LinkNode.h [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    LinkNode node;
+    tokenizer tokens(linknodetext, spacesep);
+    std::vector<unsigned int> data;
+    try {
+      std::transform(tokens.begin(), tokens.end(), std::back_inserter(data),
+                     [](const std::string &token) -> unsigned int {
+                       return boost::lexical_cast<unsigned int>(token);
+                     });
+    } catch (boost::bad_lexical_cast &) {
+      std::ostringstream errout;
```
Comment (DavidACosgrove) [inline Code/GraphMol/MolEnumerator/LinkNode.h:52]: Would this be better in a function?  There are 5 instances of pretty much the same code, with a different message.

**Review by DavidACosgrove — APPROVED** [review]
A couple of comments, but nothing substantive.

File: Code/GraphMol/MolDraw2D/MolDraw2D.cpp (hunk 1 above)
Comment (greglandrum) [reply Code/GraphMol/MolDraw2D/MolDraw2D.cpp:2372]: there's more or less no way we won't end up inside one of these if statements given that `labelPt` is initialized to `{-1000,-1000}`.

File: Code/GraphMol/MolEnumerator/LinkNode.h (hunk 2 above)
Comment (greglandrum) [reply Code/GraphMol/MolEnumerator/LinkNode.h:52]: in principle I agree with this, but when I actually do so it becomes non-obvious that there's actually an exception thrown, so the gain in brevity is offset by a loss in clarity.

**Pull request merged**